coq/coq.el

```diff
@@ -1203,21 +1203,41 @@ be called and no command will be sent to Coq."
      ;; If user issued a printing option then t printing.
      (and (string-match-p "\\(S\\|Uns\\)et\\s-+Printing" cmd)
           (> (length coq-last-but-one-proofstack) 0)))
-    (list "Show."))
+    (if coq-show-proof-stepwise
+        (or
+    (when (eq coq-diffs 'off) (list "Show." "Show Proof." ))
+    (when (eq coq-diffs 'on) (list "Show." "Show Proof Diffs."))
+    (when (eq coq-diffs 'removed) (list "Show." "Show Proof Diffs removed.")))
+   (list "Show.")))
    ((or
      ;; If we go back in the buffer and the number of abort is less than
      ;; the number of nested goals, then Unset Silent and Show the goal
      (and (string-match-p "BackTo\\s-" cmd)
           (> (length coq-last-but-one-proofstack) coq--retract-naborts)))
     ;; "Set Diffs" always re-prints the proof context with (if enabled) diffs
-    (list "Unset Silent." (if (coq--post-v810) (coq-diffs) "Show.")))
+    (list "Unset Silent." (if (coq--post-v810) (coq-diffs)
+                            (if coq-show-proof-stepwise
+    ((when (eq coq-diffs 'off)
+      "Show." "Show Proof." )
+    (when (eq coq-diffs 'on)
+      "Show." "Show Proof Diffs.")
+    (when (eq coq-diffs 'removed)
+       "Show." "Show Proof Diffs removed."))
+     "Show."))))
```

Comment:
Some code is missing in the "then" case. At first sight (modulo indentation and parens' issues) you could write:

```suggestion
    (list "Unset Silent." (append (if (coq--post-v810) (coq-diffs) "Show.")
    (when coq-show-proof-stepwise
    (when (eq coq-diffs 'off) (list "Show Proof."))
    (when (eq coq-diffs 'on) (list "Show Proof Diffs."))
    (when (eq coq-diffs 'removed) (list "Show Proof Diffs removed.")))))
```